### ISSUE TITLE
[PHP 7.2] ErrorException: Paramater must be an array ...

### ISSUE THREAD

**Issue by vrubiella** [guzzle/guzzle#1956]
| Q            | A
| ------------ | ---
| Bug?         | yes
| New Feature? | only on PHP 7.2
| Version      | 6.2.3

`ErrorException: count(): Parameter must be an array or an object that implements Countable in /www/vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:67`

CurlFactory::67
```
        if (count($this->handles) >= $this->maxHandles) {
            curl_close($resource);
        } else {
```

Maybe $this->handles is nulll, on PHP <7.2 count(null) == 0.

**Comment by vrubiella** [comment]
fixed upgrading to 6.3.0

**Comment by vrubiella** [comment]
curl_setopt_array

**Comment by vrubiella** [comment]
fixed upgrading to 6.3.0
